### PR TITLE
Convert components to individual entry points

### PR DESCRIPTION
**Pull request by cal-smith**
Closes IBM/carbon-components-angular#1320

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,158 +1,132 @@
...
+    "docs:server": "compodoc -p tsconfig.json -s -w",
+    "lint": "tslint 'src/**/*.ts'",
+    "lint:fix": "tslint --fix 'src/**/*.ts'",
+    "test": "ng test --no-watch",
+    "test:watch": "ng test --watch",
+    "build-storybook": "build-storybook -c .storybook -s .storybook/public -o dist/docs/storybook",
+    "semantic-release": "semantic-release",
+    "commit": "git-cz",
+    "ng": "ng",
+    "ng:build": "node --max_old_space_size=4096 ./node_modules/@angular/cli/bin/ng build"
```
Comment (Donisius) [inline package.json:19]: Do you still need the `--max_old_space_size=4096` if you upgrade the version of node to 12?

File: src/search/search.component.spec.ts [hunk 2]
```diff
@@ -24,14 +24,14 @@ describe("Search", () => {
...
-			}
-		})
+		// .overrideComponent(Search, {
+		// 	remove: {
+		// 		templateUrl: "search.component.html"
+		// 	},
+		// 	add: {
+		// 		template: require("./search.component.html")
+		// 	}
+		// })
```
Comment (Donisius) [inline src/search/search.component.spec.ts:34]: Is this gonna be added back in eventually?

File: package.json (hunk 1 above)
Comment (cal-smith) [reply package.json:19]: I _think_ so ... I'll double check though

File: src/search/search.component.spec.ts (hunk 2 above)
Comment (cal-smith) [reply src/search/search.component.spec.ts:34]: Probably! The tests were very unhappy with that for some reason, so I'll see if I can get it working now

**Comment by theiliad** [comment]
probably needs a more in-depth review than mine

**Comment by cal-smith** [comment]
> probably needs a more in-depth review than mine

No worries ... hopefully @zvonimirfras will have a chance to check it out ... I think @Donisius caught most of the things I forgot already

File: .storybook/preview.scss [hunk 3]
```diff
@@ -1,4 +1,5 @@
-@import '~carbon-components/scss/globals/scss/styles';
+// @import '~carbon-components/scss/globals/scss/styles';
```
[the remark this answers is not in the document]
Comment (cal-smith) [reply .storybook/preview.scss:1]: Probably not 🤔

File: angular.json [hunk 4]
```diff
@@ -1,77 +1,85 @@
...
+    "@schematics/angular:component": {
+      "prefix": "ibm",
+      "style": "css"
+    },
+    "@schematics/angular:directive": {
+      "prefix": "ibm"
+    }
+  },
+  "cli": {
+    "analytics": "738e9b32-8ce4-448f-b650-921e5656f770"
```
[the remark this answers is not in the document]
Comment (cal-smith) [reply angular.json:83]: I don't think we can 😛 that's for the Angular team ... I believe we can opt-out if we remove that block

File: src/dialog/dialog.component.ts [hunk 5]
```diff
@@ -44,7 +44,7 @@ export class Dialog implements OnInit, AfterViewInit, OnDestroy {
 	 * Maintains a reference to the view DOM element of the `Dialog`.
 	 */
 	// @ts-ignore
-	@ViewChild("dialog", { static: false }) dialog: ElementRef;
+	@ViewChild("dialog", /* TODO: add static flag */ {}) dialog: ElementRef;
```
[the remark this answers is not in the document]
Comment (cal-smith) [reply src/dialog/dialog.component.ts:47]: Yeah ... I'll have to double check, I'm not sure when that was introduced

File: src/utils/public_api.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
+export {
```
[the remark this answers is not in the document]
Comment (cal-smith) [reply src/utils/public_api.ts:1]: Yeah, this is definitely the way to go. Way cleaner than the old method ... I think eventually (if I haven't already) I'll convert everything over to the `public_api` form

File: package.json (hunk 1 above)
Comment (cal-smith) [reply package.json:19]: I'm going to leave it as is, just so we have some room in the future if something decides it's going to eat up a ton of memory ... it doesn't seem to hurt anything as is

File: src/search/search.component.spec.ts (hunk 2 above)
Comment (cal-smith) [reply src/search/search.component.spec.ts:34]: Checking it out it looks like that was a workaround from when the tests didn't know how to deal with `templateUrl` ... they do now so we can safely remove that!

**Comment by cal-smith** [comment]
Woohoo 🎉 Only thing to sort now is the checks against Angular 7/8/9 😛 `ngcc` is giving me some grief at the moment ...

**Pull request merged**